File: ios-swiftui/library/Sources/PlanetComponents/TspKeyValueLabel.swift
import SwiftUI

public struct TspKeyValueLabel: View {
    let label: String
    let value: String
    let theme: StarPlanetTheme
    public init(label: String, value: String, theme: StarPlanetTheme = .sky) {
        self.label = label
        self.value = value
        self.theme = theme
    }
    public var body: some View {
        HStack { Text(label).foregroundColor(theme.textSecondary); Spacer(); Text(value).fontWeight(.bold).foregroundColor(theme.textPrimary) }
    }
}

File: flutter/library/lib/src/tsp_key_value_label.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspKeyValueLabel extends StatelessWidget {
  const TspKeyValueLabel({super.key, required this.label, required this.value, this.theme = StarPlanetTheme.sky});
  final String label;
  final String value;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) => Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
        Text(label, style: TextStyle(color: theme.textSecondary)),
        const SizedBox(width: 16),
        Flexible(child: Text(value, textAlign: TextAlign.right, style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w800))),
      ]);
}
